Made selection better

zer0code9 opened the pull request:
- Selection for the semester is sorted by calendar (spring 2025 is before summer 2025 which is before fall 2025...)
- Only courses with the semester will show including subject and number
- Made the code easier to read

evanugarte commented on src/lib/components/Header.svelte:
```diff
@@ -25,9 +25,12 @@
 		<Search {courses} {formData} />
 	{/if}
 	<div class="menu menu-horizontal hidden gap-x-4 lg:flex">
-		<a href="/about" class="btn btn-ghost btn-sm text-lg">About</a>
```
are we gonna add the about page back? or just skip

gursheyss replied:
yea just a temp change

evanugarte commented on src/lib/components/Header.svelte:
```diff
@@ -44,6 +47,9 @@
 				<li><a href="/about" class="text-lg">About</a></li>
 				{#if user}
 					<li><Logout><button type="submit" class="text-lg">Log Out</button></Logout></li>
+				{:else if dev}
+					<li><a href="/dev-login" class="text-lg">Log In</a></li>
+					<li><a href="/dev-login" class="text-lg">Sign Up</a></li>
```
how about its just 1 button called "dev login", since they will basically lead you to the same page

evanugarte commented on src/routes/dev-login/+page.svelte:
```diff
@@ -0,0 +1,46 @@
+			}}
+		>
+			<label class="block text-sm font-medium" for="username">Username</label>
+			<input
+				id="username"
+				class="input input-secondary mt-1 w-full"
+				type="text"
+				name="username"
+				placeholder="Enter your username"
+				required
+			/>
+			<label class="block text-sm font-medium" for="email">Email</label>
+			<input
+				id="email"
+				class="input input-secondary mt-1 w-full"
+				type="email"
+				name="email"
+				placeholder="Enter your email"
+				required
+			/>
```
what if we just use 1 email input? can the username just be a copy of the email value?

evanugarte commented on src/routes/dev-login/+page.server.ts:
```diff
@@ -0,0 +1,37 @@
+			});
+		}
+
+		const session = await lucia.createSession(userId, {});
+		const sessionCookie = lucia.createSessionCookie(session.id);
```
should we even be creating a session? what does the cookie look like when we login on the browser

gursheyss replied:
it makes an auth_session cookie, we still need a local session for authenticated routes since all the user metadata is accessed directly from the session cookie, and theres really no downside to doing it like this

zer0code9 committed:
Update search.tsx

zer0code9 committed:
changed the home page to look better and can be used on a phone

zer0code9 committed:
Merge branch 'dev' into better-selection

YuraT reviewed (approved):
:fire: looks good to me

zer0code9 committed:
Updated selections and changed outline

- items are in the same y-axis as the value
- selections now have a search
- added a new search with the pick selection
- fixed icon offset
Everything works on phone

joshua-demo reviewed (approved):
lgtm
